OrbitGl/TimeGraph.cpp

```diff
@@ -848,6 +936,43 @@ std::shared_ptr<GpuTrack> TimeGraph::GetOrCreateGpuTrack(
   return track;
 }
 
+//-----------------------------------------------------------------------------
+static void SetTrackNameFromRemoteMemory(std::shared_ptr<Track> track,
+                                         uint64_t string_address,
+                                         ThreadPool* thread_pool) {
+  thread_pool->Schedule([track, string_address]() {
+    int32_t pid = Capture::GTargetProcess->GetID();
+    const auto& process_manager = GOrbitApp->GetProcessManager();
+    auto error_or_string =
+        process_manager->LoadNullTerminatedString(pid, string_address);
+
+    if (error_or_string.has_value()) {
+      GOrbitApp->GetMainThreadExecutor()->Schedule([track, error_or_string]() {
+        track->SetName(error_or_string.value());
+        track->SetLabel(error_or_string.value());
+      });
+    } else {
+      ERROR("Setting track name from remote memory: %s",
+            error_or_string.error().message());
+    }
+  });
+}
+
+//-----------------------------------------------------------------------------
+std::shared_ptr<GraphTrack> TimeGraph::GetOrCreateGraphTrack(
+    uint64_t graph_id) {
+  ScopeLock lock(m_Mutex);
+  std::shared_ptr<GraphTrack> track = graph_tracks_[graph_id];
+  if (track == nullptr) {
+    track = std::make_shared<GraphTrack>(this, graph_id);
+    SetTrackNameFromRemoteMemory(track, graph_id, GOrbitApp->GetThreadPool());
```

Comment:
Maybe pass pointer to OrbitApp to TimeGraph c-tor and use that pointer instead of the global?